src/jdk.crypto.ec/share/classes/sun/security/ec/SunEC.java

```diff
@@ -199,135 +207,185 @@ public Void run() {
 
     void putEntries() {
         HashMap<String, String> ATTRS = new HashMap<>(3);
-        ATTRS.put("ImplementedIn", "Software");
-        String ecKeyClasses = "java.security.interfaces.ECPublicKey" +
-                 "|java.security.interfaces.ECPrivateKey";
-        ATTRS.put("SupportedKeyClasses", ecKeyClasses);
-        ATTRS.put("KeySize", "256");
-
-        /*
-         *  Key Factory engine
-         */
-        putService(new ProviderService(this, "KeyFactory",
-            "EC", "sun.security.ec.ECKeyFactory",
-            List.of("EllipticCurve"), ATTRS));
-
-        /*
-         * Algorithm Parameter engine
-         */
-        // "AlgorithmParameters.EC SupportedCurves" prop used by unit test
-        boolean firstCurve = true;
-        StringBuilder names = new StringBuilder();
-
-        for (NamedCurve namedCurve :
-            List.of(
-                CurveDB.lookup("secp256r1"),
-                CurveDB.lookup("secp384r1"),
-                CurveDB.lookup("secp521r1"))) {
-            if (!firstCurve) {
-                names.append("|");
-            } else {
-                firstCurve = false;
-            }
-
-            names.append("[");
-            String[] commonNames = namedCurve.getNameAndAliases();
-            for (String commonName : commonNames) {
-                names.append(commonName);
-                names.append(",");
-            }
-
-            names.append(namedCurve.getObjectId());
-            names.append("]");
-        }
-
-        HashMap<String, String> apAttrs = new HashMap<>(ATTRS);
-        apAttrs.put("SupportedCurves", names.toString());
-
-        putService(new ProviderServiceA(this, "AlgorithmParameters",
-            "EC", "sun.security.util.ECParameters", apAttrs));
+        if (FIPSConfigurator.enableFips()) {
```

Comment:
Both branches of this `if` statement appear to be the same. Shouldn't they be different?

Reply:
same as `SunEntries.java`

Reply:
Perhaps you don't understand my question. What I see has this pattern:
```
    if (FIPSConfigurator.enableFips()) {
        /* block A */
    } else {
        /* block B */
    }
```
with block A and block B doing exactly the same thing. If this file is unchanged, we would get the same behavior. Why are you changing this file if it makes no difference?

Reply:
Its not exactly the same, the Signature engines, Key Pair Generator engine, and Key Agreement engine are disabled when FIPS is enabled.

Reply:
Ok, I see the differences now. The FIPS part is common to both branches: I suggest it be moved before an `if` statement:
```
        if (!FIPSConfigurator.enableFips()) {
            return;
        }
```
This will make it much clearer which services apply in FIPS mode or otherwise.

It would be good if you could make the distinction clearer in `SunEntries.java` as well.

Reply:
sure, I changed both `SunEntries.java` and `SunEC.java`